backend/app/assets/javascripts/spree/backend/admin.js.erb

```diff
@@ -111,8 +111,9 @@ $(document).ready(function(){
     new_table_row.find("input, select").each(function () {
       var el = $(this);
       el.val("");
-      el.prop("id", el.prop("id").replace(/\d+/, new_id))
-      el.prop("name", el.prop("name").replace(/\d+/, new_id))
+      // Replace last occurrence of a number
+      el.prop("id", el.prop("id").replace(/\d+(?=[^\d]*$)/, new_id))
+      el.prop("name", el.prop("name").replace(/\d+(?=[^\d]*$)/, new_id))
```

Comment:
Ok thanks for that explanation. This code is kind of crazy but not going to fault you for that. :)